idea/idea-fir/src/org/jetbrains/kotlin/idea/quickfix/fixes/TypeMismatchFactories.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.idea.quickfix.fixes
+
+import com.intellij.codeInsight.intention.IntentionAction
+import com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.idea.fir.api.fixes.diagnosticFixFactory
+import org.jetbrains.kotlin.idea.frontend.api.KtAnalysisSession
+import org.jetbrains.kotlin.idea.frontend.api.fir.diagnostics.KtFirDiagnostic
+import org.jetbrains.kotlin.idea.frontend.api.types.KtType
+
+object TypeMismatchFactories {
+    val argumentTypeMismatchFactory = diagnosticFixFactory<KtFirDiagnostic.ArgumentTypeMismatch> { diagnostic ->
+        getFixesForTypeMismatch(diagnostic.psi, diagnostic.expectedType, diagnostic.actualType)
+    }
+
+    val returnTypeMismatchFactory = diagnosticFixFactory<KtFirDiagnostic.ReturnTypeMismatch> { diagnostic ->
+        getFixesForTypeMismatch(diagnostic.psi, diagnostic.expected, diagnostic.actual)
```

Comment:
Ohhhhh, so parameter names we added in the diagnostics are used like this?! I know it's out of scope of this PR, but, similar to the above one (argument type mismatch), I wish these have been named as `expectedType` and `actualType`.

Reply:
Agree. We can probably clean up and make all diagnostic parameter names consistent in a separate change.

Reply:
Initiated: https://github.com/JetBrains/kotlin/pull/4370

Reply:
@punzki See if https://github.com/JetBrains/kotlin/commit/864ab2caddfaac07aaa84eb6e9a1a0e436069eb5#diff-dd09692eb8156e0b41d4532c68cc768c3fc642d7b07e79b1c49b74b38ee4835bR378-R379 affect the usage here.